### PR TITLE
Support embedding a custom registries.conf file

### PR DESCRIPTION
**Pull request by zaneb**
We need to specify a registries.conf file to set up the mirror, since the pull secret is for the mirror registry but the IPA image path points to the release image.

### CODE REVIEW

File: pkg/env/env.go [hunk 1]
```diff
@@ -9,10 +14,24 @@ type EnvInputs struct {
...
 
 func New() (*EnvInputs, error) {
 	env := &EnvInputs{}
 	err := envconfig.Process("", env)
 	return env, err
 }
+
+func (env *EnvInputs) RegistriesConf() (data []byte, err error) {
+	if env.RegistriesConfPath == "" {
+		return
```
Comment (asalkeld) [inline pkg/env/env.go:28]: not a fan of these naked returns..

**Review by asalkeld — COMMENTED** [review]
looks good, just a bit sad there is no tests :-(

**Comment by dtantsur** [comment]
Would it be possible to add tests per Angus' request?

**Comment by zaneb** [comment]
Now with tests.

File: pkg/env/env.go (hunk 1 above)
Comment (sadasu) [reply pkg/env/env.go:28]: Do we want to set an error here indicating that REGISTRIES_CONF_PATH is not set? It currently fails silently.

Comment (zaneb) [reply pkg/env/env.go:28]: I don't think we have to make it required (it doesn't fail, it just doesn't add a registry file to the ignition).

**Pull request merged**